File: plants_movil/lib/guards/islogin.guard.dart
import 'dart:async';

import 'package:flutter_modular/flutter_modular.dart';
import 'package:plants_movil/services/usuario.service.dart';

class IsLoginGuard extends RouteGuard {
  @override
  FutureOr<bool> canActivate(String path, ParallelRoute route) async {
    bool islogged = await UsuarioService().isLogin();
    if (islogged) return true;

    return false;
  }
}

File: administrador_plants/src/app/guards/is-login.guard.ts
import { Injectable } from '@angular/core';
import { ActivatedRouteSnapshot, Router, RouterStateSnapshot, UrlTree } from '@angular/router';
import { Observable } from 'rxjs';
import { UsuarioService } from '../services/usuario.service';

@Injectable({
    providedIn: 'root'
})
export class IsLoginGuard {
    constructor(private usuarioService: UsuarioService, private router: Router) {

    }
    canActivate(
        route: ActivatedRouteSnapshot,
        state: RouterStateSnapshot): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
        if (!this.usuarioService.inicioSesion()) {
            this.router.navigateByUrl("/login");
            return false;
        }

        return true;
    }

}